### PR TITLE
Fix Py_InitializeEx() initsigs parameter

### PR DESCRIPTION
**Pull request by hartcw**
I found this bug, and fixed it in my local build.
Please consider for upstreaming.

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

File: Modules/main.c [hunk 1]
```diff
@@ -1398,6 +1398,7 @@ pymain_read_conf(_PyMain *pymain, _PyCoreConfig *config,
 void
 _PyMainInterpreterConfig_Clear(_PyMainInterpreterConfig *config)
 {
+    config->install_signal_handlers = -1;
```
Comment (vstinner) [inline Modules/main.c:1401]: I don't think that it's needed, this function is only used to release memory, not to reset a config to the default config.

If you really need to reset to the default config, I would prefer to use _PyMainInterpreterConfig_INIT to reduce maintenance (avoid duplication of code).

File: Modules/main.c [hunk 2]
```diff
@@ -1441,6 +1442,8 @@ _PyMainInterpreterConfig_Copy(_PyMainInterpreterConfig *config,
 {
     _PyMainInterpreterConfig_Clear(config);
 
+    config->install_signal_handlers = config2->install_signal_handlers;
```
Comment (vstinner) [inline Modules/main.c:1445]: I would prefer to use a COPY_ATTR() macro here:

#define COPY_ATTR(ATTR) config->ATTR = config2->ATTR

Rename the existing macro to COPY_OBJ_ATTR(): see _PyCoreConfig_Copy().

**Comment by vstinner** [comment]
I created https://bugs.python.org/issue35233 to track the bug and I wrote a different fix https://github.com/python/cpython/pull/10519

My fix includes unit tests and also update test.pythoninfo.

Thanks @hartcw for spotting the bug!

**Comment by vstinner** [comment]
Thanks @hartcw, I fixed https://bugs.python.org/issue35233 in the 3.7 and master branches. I found other bugs when I wrote more unit tests :-)

**Comment by hartcw** [comment]
OK great, thanks for responding so quickly!

**Pull request closed without merging**